### PR TITLE
Make sure that editing profile without 'change mentor' permission doe…

### PR DESCRIPTION
**Pull request by MichaelKohler**
…sn't delete mentor on profile

### CODE REVIEW

**Comment by MichaelKohler** [comment]
@akatsoulas r?

File: remo/profiles/forms.py [hunk 1]
```diff
@@ -137,6 +137,16 @@ def clean_twitter_account(self):
...
+    def clean_mentor(self):
+        """Make sure that we don't save an empty mentor"""
+        has_perm_to_edit_mentor = self.request.user.has_perm('profiles.can_change_mentor')
+        cleaned_mentor = self.cleaned_data['mentor']
+        if has_perm_to_edit_mentor == False and cleaned_mentor == None:
+            initial_mentor = self.initial['mentor']
+            cleaned_mentor = User.objects.get(id=initial_mentor)
+        return cleaned_mentor
+
+
```
Comment (akatsoulas) [inline remo/profiles/forms.py]: Single line pls.

File: remo/profiles/forms.py [hunk 2]
```diff
@@ -137,6 +137,16 @@ def clean_twitter_account(self):
...
         return twitter_account.strip('@')
 
+    def clean_mentor(self):
+        """Make sure that we don't save an empty mentor"""
+        has_perm_to_edit_mentor = self.request.user.has_perm('profiles.can_change_mentor')
+        cleaned_mentor = self.cleaned_data['mentor']
+        if has_perm_to_edit_mentor == False and cleaned_mentor == None:
+            initial_mentor = self.initial['mentor']
+            cleaned_mentor = User.objects.get(id=initial_mentor)
+        return cleaned_mentor
```
Comment (akatsoulas) [inline remo/profiles/forms.py]: There is no need to query the database to get the initial mentor. You could write something like this:

```python
if not self.request.user.has_perm('profiles.can_change_mentor'):
    self.cleaned_data['mentor'] = self.instance.mentor
return self.cleaned_data['mentor']
```

File: remo/profiles/forms.py [hunk 3]
```diff
@@ -137,6 +137,16 @@ def clean_twitter_account(self):
         twitter_account = self.cleaned_data['twitter_account']
         return twitter_account.strip('@')
 
+    def clean_mentor(self):
+        """Make sure that we don't save an empty mentor"""
+        has_perm_to_edit_mentor = self.request.user.has_perm('profiles.can_change_mentor')
+        cleaned_mentor = self.cleaned_data['mentor']
+        if has_perm_to_edit_mentor == False and cleaned_mentor == None:
```
[the remark this answers is not in the document]
Comment (akatsoulas) [reply remo/profiles/forms.py]: Yes, if @MichaelKohler uses the snippet that I pasted above, this error will go away.

**Comment by MichaelKohler** [comment]
@akatsoulas done, thanks for the review. Not sure why I was under the impression that self.instance wouldn't help the old value :/

**Pull request merged**